### PR TITLE
Fix sitemap last modified date

### PR DESCRIPTION
**Pull request by aepfli**
If we do not find a proper date for a page, we assume the build date.

Signed-off-by: Simon Schrottner <simon.schrottner@dynatrace.com>

### CODE REVIEW

File: layouts/sitemap.xml [hunk 1]
```diff
@@ -2,14 +2,18 @@
 {{ $versionUsed := print "docs/" $.Site.Params.version}}
 {{ range .Data.Pages -}}
     {{ if not .Params.private -}}
-        {{ if or (in .Permalink $versionUsed) (not (in .Permalink "docs"))}}
+        {{ if or (in .RelPermalink $versionUsed) (not (in .RelPermalink "docs/"))}}
+            {{ $lastModDate := .Lastmod -}}
+            {{ if not .Lastmod -}}
+                {{ $lastModDate = now -}}
+            {{- end}}
```
Comment (mowies) [inline layouts/sitemap.xml:9]: why not put that into an if/else instead of first setting and then checking?

Comment (aepfli) [reply layouts/sitemap.xml:9]: I have yet to try to determine if this assumption is valid. Still, I usually need to define variables outside the condition anyways, so in the end, I favor just checking the error case and handling it, as it is fewer lines, and from a logic perspective, I think it is easier to follow

Comment (aepfli) [reply layouts/sitemap.xml:9]: i also coud do somethink like following based on https://gohugo.io/functions/cond/ which is actually a pretty nice and sweet idea
```suggestion
            {{ cond .Lastmod .Lastmod  now }}
```

Comment (aepfli) [reply layouts/sitemap.xml:9]: > i also coud do somethink like following based on https://gohugo.io/functions/cond/ which is actually a pretty nice and sweet idea

or
```suggestion
            {{ $lastModDate := default .Lastmod now }}
```

checking my options

Comment (mowies) [reply layouts/sitemap.xml:9]: i would prefer the `default` oneliner :)

File: layouts/sitemap.xml [hunk 2]
```diff
@@ -2,14 +2,15 @@
 {{ $versionUsed := print "docs/" $.Site.Params.version}}
 {{ range .Data.Pages -}}
     {{ if not .Params.private -}}
-        {{ if or (in .Permalink $versionUsed) (not (in .Permalink "docs"))}}
+        {{ if or (in .RelPermalink $versionUsed) (not (in .RelPermalink "docs/"))}}
+        {{ $lastModDate := default now .Lastmod}}
```
Comment (mowies) [inline layouts/sitemap.xml:6]:
```suggestion
            {{ $lastModDate := default now .Lastmod}}
```

Comment (aepfli) [reply layouts/sitemap.xml]: thank you resolved

**Review by mowies — APPROVED** [review]

**Pull request merged**